instrumentation/excon/lib/opentelemetry/instrumentation/excon/patches/socket.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Excon
+      module Patches
+        # Module to prepend to an Excon Socket for instrumentation
+        module Socket
+          private
+
+          def connect
+            return super if untraced?
+
+            if @data[:proxy]
+              conn_address = @data[:proxy][:hostname]
+              conn_port = @data[:proxy][:port]
+            else
+              conn_address = @data[:hostname]
+              conn_port = @port
+            end
+
+            attributes = { OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME => conn_address, OpenTelemetry::SemanticConventions::Trace::NET_PEER_PORT => conn_port }.merge!(OpenTelemetry::Common::HTTP::ClientContext.attributes)
+
+            if is_a?(::Excon::SSLSocket) && @data[:proxy]
+              span_name = 'HTTP CONNECT'
+              span_kind = :client
+            else
+              span_name = 'connect'
+              span_kind = :internal
+            end
+
+            tracer.in_span(span_name, attributes: attributes, kind: span_kind) do
+              super
+            end
+          end
+
+          def tracer
+            Excon::Instrumentation.instance.tracer
+          end
+
+          def untraced?
```

Comment:
This was inspired by the `Net::HTTP` instrumentation correct?

Since that time we have taken a different approach, which is to use an `untraced` helper, which is more consistent and reliable because it suppresses all child spans as opposed to potentially creating orphans: 

https://github.com/open-telemetry/opentelemetry-ruby/blob/d365dfeb3982f09cd62a17c6e67eb3792f56e460/common/lib/opentelemetry/common/utilities.rb#L90

Here is an example usage:

https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/7ba26160ec240f27bd763507ab1c3664eeb2107e/instrumentation/que/lib/opentelemetry/instrumentation/que/patches/poller.rb#L22

Reply:
I switched to using untraced instead of just calling super when we connect. Do we you recommend removing the `:untraced_hosts` check as well?

Reply:
Not at all. How do you feel about extracting a shared module function for the `untraced_hosts` check and reusing it here?

Reply:
I added a new `OpenTelemetry::Instrumentation::Concerns::UntracedHosts` module in the base directory and used that in the instrumentation class. Then, I updated the middleware and patch to call out to the instrumentation's `untraced?(host)` method. Is this what you had in mind?

Reply:
My initial though was to extract it as a module in the excon instrumentation gem but it does seem like sharing it will all instrumentations would be quite useful. 

Adding features to the Base gem gets a little tricky. Any changes we make to it must be released first and then child gems needs to be updated to have min version that supports the shared attribute. 

Since that process is a bit more challenging I would like to suggest that you isolate the shared module in this gem then we can follow up with a separate PR that moves the shared logic into the Base gem.

Reply:
Will do.

Reply:
Done, I moved the module to the excon gem with a similar path to what it would be when in the base gem.

Reply:
Nice! I'll take another look this afternoon. If all goes well I'll cut a release